File: app/src/main/java/com/dehaat/dehaatassignment/activity/BaseActivity.java
package com.dehaat.dehaatassignment.activity;


import android.content.Intent;
import android.os.Bundle;

import com.dehaat.dehaatassignment.R;

import dagger.android.support.DaggerAppCompatActivity;

public class BaseActivity extends DaggerAppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    protected void launchActivity(Class<?> cls){
        Intent intent = new Intent(this,cls);
        startActivity(intent);
    }
}

File: app/src/main/java/com/dehaat/dehaatassignment/activity/BaseActivity.kt
package com.dehaat.dehaatassignment.activity


import android.content.Intent
import android.os.Bundle

import com.dehaat.dehaatassignment.R
import com.dehaat.dehaatassignment.util.AuthUtils

import dagger.android.support.DaggerAppCompatActivity

open class BaseActivity : DaggerAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    protected fun launchActivity(cls: Class<*>) {
        val intent = Intent(this, cls)
        startActivity(intent)
    }

    protected fun logout() {
        AuthUtils.setLoggedInPreference(applicationContext, false)
        launchActivity(LoginActivity::class.java)
        finish()
    }
}
